### PR TITLE
Recycling center

### PR DESCRIPTION
**Pull request by itanasi**
Adds Recycling Center
Blocks showing the building in the choices if 5 or more are built
If 5 are building, still shows but with a Rejection Text
Image included thanks to @SimonCeder. Credit added to Credits.md

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/Building.kt [hunk 1]
```diff
@@ -454,6 +454,12 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
     override fun shouldBeDisplayed(cityConstructions: CityConstructions): Boolean {
         if (cityConstructions.isBeingConstructedOrEnqueued(name))
             return false
+        if (hasUnique(UniqueType.MaxNumberBuilding)) {
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/Building.kt:457]: Unnecessary if

File: android/assets/jsons/Civ V - Gods & Kings/Buildings.json [hunk 2]
```diff
@@ -1049,6 +1049,12 @@
 		"requiredTech": "Ecology",
 		"uniques": ["Must be next to [Desert]", "Cannot be built with [Nuclear Plant]"]
 	},
+	{
+		"name": "Recycling Center",
+		"maintenance": 3,
+		"requiredTech": "Ecology",
+		"uniques": ["Provides [2] [Aluminum]","Can have [5] of this building in your empire at most"]
```
Comment (yairm210) [inline android/assets/jsons/Civ V - Gods & Kings/Buildings.json:1056]: Change to "Limited to [] per Civilization" so we can reuse this for units, improvements etc

**Comment by itanasi** [comment]
Resolved requests. Added code to destroy building if exceeds max allowed (including enqueued). Tested in Human v Human game both captured and traded.

Future improvement: perhaps remove (and refund) queued buildings first before resorting to destroying the new one in the captured/traded city. But that's a bit complicated for today's work

File: core/src/com/unciv/models/ruleset/Building.kt (hunk 1 above)
Comment (itanasi) [reply core/src/com/unciv/models/ruleset/Building.kt]: Done

**Pull request merged**